app/main.py

```diff
@@ -1 +1,47 @@
-# write your code here
+class Animal:
+
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+        # Animal.alive.append(self.__dict__)
+        print(Animal.alive, len(Animal.alive))
+
+    # def __str__(self):
+    #     res = []
+    #     for _ in self.alive:
+    #         res.append(self.__dict__)
+    #     Animal.alive = res
+    #     return Animal.alive
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        if self.hidden is False:
+            self.hidden = True
+        else:
+            self.hidden = False
+
+
+class Carnivore(Animal):
+
+    def bite(self, other: object):
+        if other.hidden == False and type(other) != Carnivore:
```

Comment:
Use `isinstance` instead of `type` for checking type of object

Reply:
oK